### PR TITLE
Fix mixed media file extensions upload

### PR DESCRIPTION
**Pull request by Dens49**
1739954130

Fixes a bug introduced in #42 (not included in any release).
Mixed file extensions led to mixing up the mapping between uploaded image files and medias.

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -180,28 +207,65 @@ def test_upload_media_files_with_presigned_urls_with_multiple_file_extensions(
...
     # for every call
+    file_paths = {
+        0: "./my_test_media_1.jpg",
+        1: "./my_test_media_2.png",
+        2: "./my_test_media_3.jpg",
+        3: "./my_test_media_4.png",
+    }
+
+    def get_presigned_media_upload_url_side_effect_function(
+        dataset_id, file_extension, batch_size
```
Comment (katia-qm) [inline tests/test_client.py:218]: nitpick: unused dataset id

File: tests/test_client.py [hunk 2]
```diff
@@ -180,28 +207,65 @@ def test_upload_media_files_with_presigned_urls_with_multiple_file_extensions(
...
+    ):
+        response = []
+        for i in range(batch_size):
+            # creates idx 1, 2 in the first call, and 3, 4 in the second call of the method
+            idx = i + get_presigned_media_upload_url_spy.call_count
+            if get_presigned_media_upload_url_spy.call_count > 1:
+                idx += 1
```
Comment (katia-qm) [inline tests/test_client.py:225]: I don't really get how it works, but I trust you:)

**Review by katia-qm — APPROVED** [review]
Great job and thanks for the thorough tests!

File: tests/test_client.py (hunk 2 above)
Comment (Dens49) [reply tests/test_client.py:225]: step through it in the debugger, then it should be self evident, but it's really not worth the time

**Pull request merged**